src/components/SideMenu.js

```diff
@@ -0,0 +1,147 @@
+import {
+  AppBar,
+  Divider,
+  Drawer,
+  Hidden,
+  IconButton,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Toolbar,
+  Typography
+} from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import {
+  Dashboard as DashboardIcon,
+  Menu as MenuIcon
+} from '@material-ui/icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Link } from 'react-router-dom';
+import AuthButton from './AuthButton';
+const drawerWidth = 240;
+
+const styles = theme => ({
+  root: {
+    display: 'flex'
+  },
+  drawer: {
+    [theme.breakpoints.up('sm')]: {
+      width: drawerWidth,
+      flexShrink: 0
+    }
+  },
+  appBar: {
+    zIndex: theme.zIndex.drawer + 1
+  },
+  menuButton: {
+    marginRight: 20,
+    [theme.breakpoints.up('sm')]: {
+      display: 'none'
+    }
+  },
+  toolbar: theme.mixins.toolbar,
+  drawerPaper: {
+    width: drawerWidth
+  },
+  content: {
+    flexGrow: 1,
+    padding: theme.spacing.unit * 3
+  }
+});
+
+class SideMenu extends React.Component {
+  state = {
+    mobileOpen: false
+  };
+
+  handleDrawerToggle = () => {
+    this.setState(state => ({ mobileOpen: !state.mobileOpen }));
+  };
+
+  render() {
+    const { classes, theme, children } = this.props;
+
+    const drawer = (
+      <div>
+        <Hidden smDown>
+          <div className={classes.toolbar} />
+        </Hidden>
+        <Divider />
+        <List>
+          <Link to="/" style={{ textDecoration: 'none' }} onClick={this.ha}>
```

Comment:
try to avoid inline styles
put them into a seperate .scss files and use classes (even for small things like these)